core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -329,7 +330,20 @@ public Result visit(Filter e) {
           ImmutableSet.of(Clause.HAVING));
       parseCorrelTable(e, x);
       final Builder builder = x.builder(e);
-      builder.setHaving(builder.context.toSql(null, e.getCondition()));
+      SqlNode condition = builder.context.toSql(null, e.getCondition());
+      if (ImmutableBitSet.union(aggregate.getGroupSets()).cardinality()
+          != aggregate.getGroupSet().cardinality()) {
```

Comment:
may "!=" replaced by "<" be better?

Reply:
use !ImmutableBitSet.union(aggregate.getGroupSets()).equals(aggregate.getGroupSet()) repalce.